CRM/Price/BAO/LineItem.php

```diff
@@ -455,7 +455,7 @@ public static function processPriceSet($entityId, $lineItem, $contributionDetail
           // CRM-19094: entity_table is set to civicrm_membership then ensure
           // the entityId is set to membership ID not contribution by default
           elseif ($line['entity_table'] == 'civicrm_membership' && !empty($line['entity_id']) && $line['entity_id'] == $contributionDetails->id) {
-            $membershipId = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_MembershipPayment', 'contribution_id', $line['entity_id'], 'membership_id');
+            $membershipId = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_MembershipPayment', $line['entity_id'], 'membership_id', 'contribution_id');
```

Comment:
This why I prefer the api - the more I look at this function sig the less sure I am about what the vars mean :-)

getFieldValue($daoName, $searchValue, $returnColumn = 'name', $searchColumn = 'id', $force = FALSE) {

So $searchValue is the contribution id & you have the line['entity_id'] which may or may not be the same
$returnColumn you have correctly changed to membership_id
$searchColumn you have correctly changed to contribution_id

This seems like a potentially critical bug & it's worth merging - @monishdeb is likely not around this week but perhaps he can do a test as a follow up

Reply:
I think we can do a unit test directly on that function

Reply:
Improved parameter names as per https://chat.civicrm.org/civicrm/pl/cwqzxf99sbgibcbr836rraijha